### PR TITLE
vaultwarden: update to 1.31.0; vaultwarden-web: update to 2024.5.1b.

### PR DESCRIPTION
**Pull request by TinfoilSubmarine**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: common/build-style/cargo.sh [hunk 1]
```diff
@@ -5,13 +5,17 @@
...
+
+	${make_cmd} build --locked --target ${RUST_TARGET} ${configure_args}
 }
 
 do_check() {
 	: ${make_cmd:=cargo auditable}
 
-	${make_check_pre} ${make_cmd} test --release --locked --target ${RUST_TARGET} \
+	${make_check_pre} ${make_cmd} test --locked --target ${RUST_TARGET} \
 		${configure_args} ${make_check_args}
```
Comment (ahesford) [inline common/build-style/cargo.sh:19]: I think this change merits discussion in a separate PR.

File: srcpkgs/vaultwarden/template [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
 
 make_dirs="/var/lib/vaultwarden 0750 _vaultwarden _vaultwarden"
 
+# fails on i686, see https://github.com/dani-garcia/vaultwarden/issues/4320
+if [ "$XBPS_TARGET_MACHINE" = "i686" ]; then
+	configure_args+=" --profile=release-low"
+	do_build() {
+		: ${make_cmd:=cargo auditable}
+
+		${make_cmd} build --locked --target ${RUST_TARGET} ${configure_args}
```
Comment (ahesford) [inline srcpkgs/vaultwarden/template:29]:
```suggestion
		cargo auditable build --locked --target ${RUST_TARGET} ${configure_args}
```

File: srcpkgs/vaultwarden/template [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
+	}
+	do_check() {
+		: ${make_cmd:=cargo auditable}
+
+		${make_check_pre} ${make_cmd} test --locked --target ${RUST_TARGET} \
+			${configure_args} ${make_check_args}
```
Comment (ahesford) [inline srcpkgs/vaultwarden/template:35]:
```suggestion
		cargo auditable test --locked --target ${RUST_TARGET} \
			${configure_args} ${make_check_args}
```

**Review by ahesford — CHANGES_REQUESTED** [review]
There's no need for indirection with the function overrides.

**Comment by TinfoilSubmarine** [comment]
> There's no need for indirection with the function overrides.

fixed

**Comment by ahesford** [comment]
Thanks!

**Pull request closed without merging**